frontend/app/services/SubscriptionService.scala

```diff
@@ -105,76 +106,76 @@ object SubscriptionService {
 class SubscriptionService(val tierPlanRateIds: Map[ProductRatePlan, String],
                           val zuoraSoapService: ZuoraSoapService,
                           val zuoraRestService: ZuoraRestService) extends AmendSubscription with LazyLogging {
-  import SubscriptionService._
-
-  def getAccount(memberId: MemberId): Future[Account] =
-    zuoraSoapService.query[Account](s"crmId='${memberId.salesforceAccountId}'").map(sortAccounts(_).last)
-
-  //TODO before we do subs we need to filter by rate plans membership knows about
-  def getSubscriptions(memberId: MemberId): Future[Seq[Subscription]] = {
-    for {
-      account <- getAccount(memberId)
-      subscriptions <- zuoraSoapService.query[Subscription](s"AccountId='${account.id}'")
-    } yield subscriptions
-  }
-
-  def memberTierFeatures(memberId: MemberId): Future[Seq[Rest.Feature]] =
-    for {
-      account <- getAccount(memberId)
-      features <- zuoraRestService.productFeaturesByAccount(account.id)
-    } yield features
 
-  def getSubscription(subscriptionId: String): Future[Subscription] = zuoraSoapService.queryOne[Subscription](s"Id='$subscriptionId'")
-
-  def getSubscriptionStatus(memberId: MemberId): Future[SubscriptionStatus] = {
-    for {
-      subscriptions <- getSubscriptions(memberId)
-
-      if subscriptions.nonEmpty
-
-      where = subscriptions.map { sub => s"SubscriptionId='${sub.id}'" }.mkString(" OR ")
-      amendments <- zuoraSoapService.query[Amendment](where)
-    } yield {
-      val latestSubscriptionId = sortSubscriptions(subscriptions).last.id
+  import SubscriptionService._
+  val MembershipProductType = "Membership"
+
+  private def subscriptionVersions(subscriptionNumber: String): Future[Seq[Subscription]] = for {
+    subscriptions <- zuoraSoapService.query[Subscription](s"Name = '$subscriptionNumber'")
+  } yield subscriptions
+
+  def accountWithLatestMembershipSubscription(member: MemberId): Future[(Account, Rest.Subscription)] = for {
+    accounts <- zuoraSoapService.query[Account](s"crmId='${member.salesforceAccountId}'")
```

Comment:
Also, notice that `query` and `queryOne` are pretty much the only public method in that class